src/gluonts/core/component.py

```diff
@@ -506,6 +508,21 @@ def get_mxnet_context(gpu_number=0) -> mx.Context:
         return mx.context.cpu()
 
 
+def normalize_ctx(
+    ctx: Optional[Union[str, mx.Context, List[mx.Context]]]
+) -> List[mx.Context]:
+    if isinstance(ctx, str):
+        return [mx.Context(ctx)]
+    if ctx is not None:
+        return [ctx] if not isinstance(ctx, list) else ctx
+    else:
+        return (
+            [mx.gpu(i) for i in range(mx.context.num_gpus())]
+            if mx.context.num_gpus()
+            else [mx.cpu()]
```

Comment:
I think I prefer this:

```python
def normalize_ctx(
    ctx: Optional[Union[str, mx.Context, List[mx.Context]]]
) -> List[mx.Context]:
    if ctx is None:
        if mx.context.num_gpus():
            return [mx.gpu(i) for i in range(mx.context.num_gpus())]
        else:
            return [mx.cpu()]

    if isinstance(ctx, str):
        return [mx.Context(ctx)]

    if isinstance(ctx, list):
        return ctx

    return [ctx]
```

Reply:
No sure the difference! would it make it more readable?

Reply:
Yes, handling `None` first, then the other cases and all in one level.